dist/fabric.js

```diff
@@ -6850,8 +6850,11 @@ fabric.ElementsParser = function(elements, callback, options, reviver, parsingOp
     var target = transform.target, localPoint = getLocalPoint(transform, transform.originX, transform.originY, x, y),
         strokePadding = target.strokeWidth / (target.strokeUniform ? target.scaleX : 1),
         multiplier = isTransformCentered(transform) ? 2 : 1,
-        oldWidth = target.width, hasResized,
-        newWidth = Math.abs(localPoint.x * multiplier / target.scaleX) - strokePadding;
+        oldWidth = target.width, hasResized;
+    var calcWidth = localPoint.x * multiplier / target.scaleX;
+    var newWidth = (target.lockScalingFlip ?
+      Math.max(calcWidth, target.padding) : Math.abs(calcWidth)) - strokePadding;
```

Comment:
Need to use a Math.max with the target.padding to avoid a weird wiggle that occurs because the getLocalPoint function uses that as well.